yb-voyager/cmd/conflictDetectionCache.go

```diff
@@ -0,0 +1,101 @@
+package cmd
```

Comment:
A simpler implementation using mutex-and-condition-variable pair:
```
package cmd

import (
	"sync"

	log "github.com/sirupsen/logrus"
	"github.com/yugabyte/yb-voyager/yb-voyager/src/tgtdb"
)

type ConflictDetectionCache struct {
	sync.Mutex
	m    map[int64]*tgtdb.Event
	cond *sync.Cond
}

func NewConflictDetectionCache() *ConflictDetectionCache {
	c := &ConflictDetectionCache{}
	c.m = make(map[int64]*tgtdb.Event)
	c.cond = sync.NewCond(&c.Mutex)
	return c
}

func (c *ConflictDetectionCache) Put(event *tgtdb.Event) {
	c.Lock()
	defer c.Unlock()
	c.m[event.Vsn] = event
}

func (c *ConflictDetectionCache) RemoveEvents(batch *tgtdb.EventBatch) {
	c.Lock()
	defer c.Unlock()
	eventRemoved := false
	for _, event := range batch.Events {
		if _, ok := c.m[event.Vsn]; ok {
			delete(c.m, event.Vsn)
			eventRemoved = true
		}
	}
	if eventRemoved {
		c.cond.Broadcast()
	}
}

func (c *ConflictDetectionCache) WaitUntilConflict(event *tgtdb.Event) {
	c.Lock()
	defer c.Unlock()
retry:
	for _, ev := range c.m {
		if c.EventsConflict(event, ev) {
			log.Infof("Waiting for event %d to complete before applying event %d", ev.Vsn, event.Vsn)
			c.cond.Wait()
			goto retry
		}
	}
}

func (c *ConflictDetectionCache) EventsConflict(a, b *tgtdb.Event) bool {
	// TODO: implement
	return false
}

```

Reply:
done